src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,92 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> components = new HashMap<>();
+
+    private Injector() {
+        try {
+            registerComponents();
+            injectDependencies();
+        } catch (Exception e) {
+            throw new RuntimeException("Error initializing injector", e);
+        }
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    private void registerComponents() {
+        Class<?>[] classes = {
+                ProductServiceImpl.class,
+                ProductParserImpl.class,
+                FileReaderServiceImpl.class
+        };
+
+        for (Class<?> clazz : classes) {
+            if (clazz.isAnnotationPresent(Component.class)) {
+                Constructor<?> constructor;
+                try {
+                    constructor = clazz.getDeclaredConstructor();
+                } catch (NoSuchMethodException e) {
+                    throw new RuntimeException("Could not find constructor for "
+                            + clazz.getName(), e);
```

Comment:
When throwing an exception, add an informative message to it. Also, don't forget to add an exception you're catching in the `catch` block to the `RuntimeException` object you throw. This is implemented correctly here, but ensure to follow this pattern consistently.